lib/maplibre-compose/src/androidMain/kotlin/dev/sargunv/maplibrecompose/core/layer/SymbolLayer.kt

```diff
@@ -33,245 +38,243 @@ internal actual class SymbolLayer actual constructor(id: String, source: Source)
   }
 
-  actual fun setIconAnchor(anchor: Expression<SymbolAnchor>) {
+  actual fun setIconAnchor(anchor: EnumExpression<SymbolAnchor>) {
     impl.setProperties(PropertyFactory.iconAnchor(anchor.toMLNExpression()))
   }
 
-  actual fun setIconPitchAlignment(pitchAlignment: Expression<IconPitchAlignment>) {
+  actual fun setIconPitchAlignment(pitchAlignment: EnumExpression<IconPitchAlignment>) {
     impl.setProperties(PropertyFactory.iconPitchAlignment(pitchAlignment.toMLNExpression()))
   }
 
-  actual fun setIconOpacity(opacity: Expression<Number>) {
+  actual fun setIconOpacity(opacity: FloatExpression) {
     impl.setProperties(PropertyFactory.iconOpacity(opacity.toMLNExpression()))
   }
 
-  actual fun setIconColor(color: Expression<Color>) {
+  actual fun setIconColor(color: ColorExpression) {
     impl.setProperties(PropertyFactory.iconColor(color.toMLNExpression()))
   }
 
-  actual fun setIconHaloColor(haloColor: Expression<Color>) {
+  actual fun setIconHaloColor(haloColor: ColorExpression) {
     impl.setProperties(PropertyFactory.iconHaloColor(haloColor.toMLNExpression()))
   }
 
-  actual fun setIconHaloWidth(haloWidth: Expression<Dp>) {
+  actual fun setIconHaloWidth(haloWidth: DpExpression) {
     impl.setProperties(PropertyFactory.iconHaloWidth(haloWidth.toMLNExpression()))
   }
 
-  actual fun setIconHaloBlur(haloBlur: Expression<Dp>) {
+  actual fun setIconHaloBlur(haloBlur: DpExpression) {
     impl.setProperties(PropertyFactory.iconHaloBlur(haloBlur.toMLNExpression()))
   }
 
-  actual fun setIconTranslate(translate: Expression<DpOffset>) {
+  actual fun setIconTranslate(translate: DpOffsetExpression) {
     impl.setProperties(PropertyFactory.iconTranslate(translate.toMLNExpression()))
   }
 
-  actual fun setIconTranslateAnchor(translateAnchor: Expression<TranslateAnchor>) {
+  actual fun setIconTranslateAnchor(translateAnchor: EnumExpression<TranslateAnchor>) {
     impl.setProperties(PropertyFactory.iconTranslateAnchor(translateAnchor.toMLNExpression()))
   }
 
-  actual fun setTextPitchAlignment(pitchAlignment: Expression<TextPitchAlignment>) {
+  actual fun setTextPitchAlignment(pitchAlignment: EnumExpression<TextPitchAlignment>) {
     impl.setProperties(PropertyFactory.textPitchAlignment(pitchAlignment.toMLNExpression()))
   }
 
-  actual fun setTextRotationAlignment(rotationAlignment: Expression<TextRotationAlignment>) {
+  actual fun setTextRotationAlignment(rotationAlignment: EnumExpression<TextRotationAlignment>) {
     impl.setProperties(PropertyFactory.textRotationAlignment(rotationAlignment.toMLNExpression()))
   }
 
-  actual fun setTextField(field: Expression<TFormatted>) {
+  actual fun setTextField(field: FormattedExpression) {
     impl.setProperties(PropertyFactory.textField(field.toMLNExpression()))
   }
 
-  actual fun setTextFont(font: Expression<List<String>>) {
+  actual fun setTextFont(font: ListExpression<StringExpression>) {
```

Comment:
By the way, this *looks* odd. It reads like this is going to evaluate to `List<StringExpression>`

Reply:
If the native code allows the list to be produced by some expression (like getting it from feature properties) then we can't do `List<StringExpression>`

Reply:
Yes, I just stated that it *looks* odd, nevermind